apis/kueue/v1alpha1/multikueueconfig_types.go

```diff
@@ -53,6 +53,14 @@ type KubeConfig struct {
 type MultiKueueClusterSpec struct {
 	// Information how to connect to the cluster.
 	KubeConfig KubeConfig `json:"kubeConfig"`
+
+	// A label value used to track the creator of workloads in the worker cluster.
+	//
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="Value is immutable"
+	// +kubebuilder:validation:MaxLength=63
+	// +kubebuilder:default=multikueue
+	GarbageCollectionKey string `json:"garbageCollectionKey,omitempty"`
```

Comment:
Does the label key need to be configurable? Does it bring any more value that just true / false? I see the value is taken from the controllerName, but this is hard-coded for multikueue.

Reply:
It could be useful.

An alternative to the new API field could be a kueue config key. (We'll need a multikueue section to make the interval on which the GC runs configurable anyway).

@alculquicondor  WDYT?

Reply:
I think it could be useful, but I wouldn't make it just about GC. It can be useful more generally, just to identify the jobs in the worker clusters.

I'm fine starting with no configuration too.

Reply:
From multikueue internals perspective it's configurable and there is no point in hard-coding  the value now.

The question in my opinion is if we want to configure it at a MultiKueueCluster level or at a kueue config level (like the GCTimeout).